components/Common/Select/index.tsx

```diff
@@ -59,7 +60,17 @@ const Select: FC<SelectProps> = ({
                       value={value}
                       className={`${styles.item} ${styles.text}`}
                     >
-                      <Primitive.ItemText>{label}</Primitive.ItemText>
+                      <Primitive.ItemText>
+                        {typeof item !== 'string' && item.iconImageUrl && (
```

Comment:
I think it's easier if the `values` is a `string[]` that you transform them to {} objects... So pretty much:

```tsx
const mappedValues = useMemo(() => values.map(value => typeof value === 'string' ? ({ value, label: value }) : item), [values]);

mappedValues.map(item => <Primitive.Item>...</Primitive.Item>);
```

This reduces the need for adding extra checks everywhere

Reply:
(the 2nd snippet you can just copy past and replace IMO)

Reply:
We were saved from performing various tricks and checks within the component. It's definitely much more readable, thank you!